src/main/resource-applier/resource-applier.ts

```diff
@@ -49,23 +57,17 @@ export class ResourceApplier {
       "-o", "json",
     );
 
-    try {
-      const { stdout } = await promiseExecFile(kubectlPath, args);
-
-      return stdout;
-    } catch (error) {
-      if (isChildProcessError(error)) {
-        throw error.stderr ?? error;
-      }
+    const result = await this.dependencies.execFile(kubectlPath, args);
 
-      throw error;
+    if (result.callWasSuccessful) {
+      return result.response;
     }
+
+    throw result.error.stderr || result.error.error;
```

Comment:
Again, this doesn't make sense. `execFile` should abstract this type of complexity away from usages. Best would be that we define error as an interface, e.g.

```typescript 
interface Error { 
  code: string;
  message: string;
  
  original: unknown // Error, string, whatever
} 
```

In this way, we could make `AsyncResult`

```typescript
type AsyncResult<Response> =  { callWasSuccessful: true, response: Response } | { callWasSuccessful: false, error: Error }
```